### DISCUSSION TITLE
Entangle issue in multi dimensional array property

### DISCUSSION THREAD

**Discussion by Joemires** [livewire/livewire#7059] [1. Help]
```php
<?php


use function Livewire\Volt\{layout, state, mount};
use function Laravel\Folio\{name, middleware};

name('manage.preference');

middleware(['auth', 'verified', 'tenancy']);

state(['business' => fn () => auth()->business(), 'business_hours' => []]);

state(['opened_from', 'closed_at']);

mount(function () {
    /** $this->business_hours = collect(['monday' => [], 'tuesday' => [], 'wednessday' => [], 'thursday' => [], 'friday' => [], 'saturday' => [], 'sunday' => []])
        ->merge(data_get($this->business->meta, 'opening_hours'));
   ***/
    $this->business_hours = ['monday' => ['12:30pm - 3:00pm']];
}); 

$saveBusinessHours = function () {
    /** $this->business->update([
        'meta->opening_hours' => $this->business_hours
    ]); **/

    $this->reset('opened_from', 'closed_at');
}
?>

@volt
<form wire:submit="saveBusinessHours" x-data="{ days: @entangle('business_hours') }" x-init="$watch('days', (v) => console.log(v)); console.log(days)">
    <div class="px-4 py-5 bg-white dark:bg-gray-800 sm:p-6 shadow sm:rounded-tl-md sm:rounded-tr-md">
        <ul class="text-gray-400">
            <template x-for="(hours, day) in days">
                <li>
                    <span x-text="day"> </span> -
                    <template x-for="hour in hours">
                        <i x-text="hour"></i>
                    </template>
                </li>
            </template>
        </ul>
    </div>

    <div class="flex items-center justify-end px-4 py-3 bg-gray-50 dark:bg-gray-800 text-right sm:px-6 shadow sm:rounded-bl-md sm:rounded-br-md">
        <x-primary-button>{{ __('Save Bank') }}</x-primary-button>
    </div>

    <x-dialog id="custom" title="Add a set of hour" description="Select an opening and closing hour, click and save it to continue">
        <div class="mt-4 grid grid-cols-2 gap-2">
            <x-time-picker label="Opened From" placeholder="Opening time" id="opened_from" wire:model.defer="opened_from" />
            <x-time-picker label="Closed At" placeholder="Closing time" id="closed_at" wire:model.defer="closed_at" />
        </div>
    </x-dialog>
</form>
@endvolt
```
The above code would render this
![image](https://github.com/livewire/livewire/assets/34143261/ee88ba2f-0e37-4b16-b4ab-5563dff9c30b)

The issue I am facing is tht when the submit button is clicked at first, There's an error of ```hour is not defined``` depending on the length of hours available in the business_hours property
![image](https://github.com/livewire/livewire/assets/34143261/0c296b7a-0fde-423a-bbc5-bc5015ed01eb)

On clicking the second time, we now have the hour rendered two times
![image](https://github.com/livewire/livewire/assets/34143261/a3cc51db-0bde-447f-b1a9-caba64270b78)

**Comment by lrljoe** [comment]
Define defaults, or code to expect blank arrays .

Use appropriate validation and merging

You need to write application logic for determining clashes etc.
PHP/Laravel/Livewire will not automagically do schedule merging, just because you have a day of the week in an array...

**Reply by Joemires** [reply]
I did apply default if you look at the commented code, I just commented it incase someone wants to try it out as well

The error I'm getting is if I try submitting the business hour form, I get an undefined hour and an empty badge is added to it
On submitting the second time, a replicate of my available values is appended again
But refreshing still keeps the available days I had irrespective of the duplicate badges

**Comment by gofish543** [comment]
Tis still an issue
